set-version.sh

```diff
@@ -1,24 +1,32 @@
 #!/bin/bash -e
 
 NEW_VERSION=$1
+NEW_NPM_VERSION=${2:-${1}}
```

Comment:
I still think we should not be using a seperate version for NPM. Instead the input versions should all adhere to valid semantic versions.

Reply:
This line has nothing to do with using valid semantic versions. It allows our downstream build script to override the NPM version if needed (mainly for debug/investigation purpose).
Even if one uses set-version.sh with one parameter and specifies "non-semantic" version it will use it anyway (no validation of version is present).
Our production build script is using "./set-version.sh use_current".

Reply:
Why would it need to do that? Wouldn't you just override all the versions?

Reply:
The internal build system assigns version to keycloak (and aligns dependencies) before start of maven build.
We have to use what was assigned. The problem is that assigned version is Maven compatible, but not npm compatible.
Therefore we need two of them. See my comment in set-version.sh script.

Reply:
> Therefore we need two of them. See my comment in set-version.sh script.

Or the version can be a valid semantic version,  in which case it would work for both.